src/libcore/result.rs

```diff
@@ -521,14 +521,16 @@ impl<T, E> Result<T, E> {
         }
     }
 
-    /// Applies a function to the contained value (if any),
-    /// or returns the provided default (if not).
+    /// Applies a function to the contained value (if [`Ok`]),
+    /// or returns the provided default (if [`Err`]).
     ///
     /// Arguments passed to `map_or` are eagerly evaluated; if you are passing
     /// the result of a function call, it is recommended to use [`map_or_else`],
     /// which is lazily evaluated.
     ///
     /// [`map_or_else`]: #method.map_or_else
+    /// [`Ok`]: enum.Result.html#variant.Ok
+    /// [`Err`]: enum.Result.html#variant.Err
```

Comment:
This can be simplified:

```suggestion
    /// Applies a function to the contained value (if [`Ok`][]),
    /// or returns the provided default (if [`Err`][]).
    ///
    /// Arguments passed to `map_or` are eagerly evaluated; if you are passing
    /// the result of a function call, it is recommended to use [`map_or_else`],
    /// which is lazily evaluated.
    ///
    /// [`map_or_else`]: #method.map_or_else
```

`Ok` and `Err` are in scope so rustdoc will generate the links for you.

Reply:
oh wait I have to change the things you mentioned myself?

Reply:
do you want to make the change? this is in a rollup so let me know quickly xD

Reply:
well its unnecessary. Only if needed